timemachine/md/minimizer.py

```diff
@@ -326,8 +326,8 @@ def local_minimize(x0, val_and_grad_fn, local_idxs):
     x_local_shape = (len(local_idxs), 3)
     u_0, _ = val_and_grad_fn(x0)
 
-    # deal with overflow
-    guard_threshold = 1e6
+    # deal with overflow, empirically obtained by testing on some real systems.
+    guard_threshold = 5e4
```

Comment:
Possible alternatives to consider in the future:
* Detect possible precision loss in more cases by also inspecting the gradient `norm(grad(f)(x))` rather than only the function values `abs(f(x))` or `abs(f(x) - f(x0))` or similar
* This can be interpreted as restricting the search to the set {x : abs(f(x) - f(x0)) < 5e4}. Could make this constraint more restrictive, e.g. by also enforcing small distance from x0.

Possible investigation in the future:
* Reproduce this issue as a possible symptom of https://github.com/proteneer/timemachine/issues/481 , since violation of line search assumptions is likely a canary for self-inconsistency of function, gradient implementations at some values of x.